aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/ApplicationLifeCycleHelper.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.analytics.pinpoint;
+
+import android.app.Activity;
+import android.app.Application;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.content.IntentFilter;
+import android.os.Bundle;
+import android.util.Log;
+
+import com.amazonaws.mobile.auth.core.internal.util.ThreadUtils;
```

Comment:
It would be better to use a utility that exists in Amplify, directly. This utility is from a dependency which may/may not be removed in the future (when we do the auth category), and also it is marked `.internal.` (meaning, "don't consume this from another library.")